src/commands/verify.ts

```diff
@@ -50,39 +42,61 @@ export async function execute(interaction: ChatInputCommandInteraction) {
     const idConfirmationMuniParsedUrl = new URL(
         sanitizeUrl(idConfirmationMuni)
     );
-    for (const key in userLogJSON) {
-        if (interaction.user.id === userLogJSON[key].id) {
-            return interaction.reply({
-                content:
-                    'User already verified! Contact admin if you need to verify again.',
-                ephemeral: true,
-            });
-        } else if (
-            userLogJSON[key].idThesis ===
-            bachelorThesisParsedUrl.pathname.split('/')[3]
-        ) {
-            return interaction.reply({
-                content: 'This thesis is already used! Please contact admin.',
-                ephemeral: true,
-            });
-        }
+
+    await interaction.reply({
+        content: 'Verifying, wait please...',
+        ephemeral: true,
+    });
+    let user;
+    try {
+        user = await prisma.users.findMany({
+            where: {
+                discordId: interaction.user.id,
+            },
+        });
+    } catch (err) {
+        console.log(`Database error: ${err}`);
+        return interaction.editReply({
+            content: 'Verification failed! Contact admin.',
+        });
+    }
+    if (user.length != 0) {
+        return interaction.editReply({
+            content:
+                'User already verified! Contact admin if you need to verify again.',
```

Comment:
Shouldn't there be a possibility for user to reverify for some reason (new account/change of spec)? But in that case, we would need to remove the previous verification where this thesis was used.

Reply:
Resolved in latest commit.